x-pack/legacy/plugins/maps/public/layers/fields/field_with_origin.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import { AbstractField } from './field';
+
+export class FieldWithOrigin extends AbstractField {
```

Comment:
Is `FieldWithOrigin` needed? Why not just have AbstractField set origin to FIELD_ORIGIN.UNDEFINED when a value is not provided?

Reply:
Removed the class. All terms are considered 'join' fields.